### ISSUE TITLE
[ntuple] RNTuple error when serializing TClass typedef members

### ISSUE THREAD

**Issue by mxxo** [root-project/root#7861]
- [x] Checked for duplicates

### Describe the bug

TClass members which are typedefs of supported RNTuple types are not being resolved to their true types during serialization, which is causing runtime errors.

### To Reproduce

Define a custom class with, e.g. a typedef'd `std::string` member intended to be serialized using rootcling and `TClass`
```cpp
class CustomStruct { 
   typedef std::string typedef_string;
   typedef_string str;
};

auto classField = RField<CustomStruct>("field");
```
Trying to create this field fails at run time:
```
C++ exception with description "Field npos has unknown type const string::size_type (unchecked RResult access!)
At:
  static ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> >
ROOT::Experimental::Detail::RFieldBase::Create(const string&, const string&) 
[/home/max/projects/rootdev/root/tree/ntuple/v7/src/RField.cxx:202]
```
### Setup
ROOT master

### Additional context
This is likely resolved by using `GetTrueTypeName` instead of `GetFullTypeName` here (thank you @jblomer):
https://github.com/root-project/root/blob/28f86b343f4ec280937fbbe8ecf64d9582651dcb/tree/ntuple/v7/src/RField.cxx#L550

**Comment by jalopezg-r00t** [comment]
I noticed today that this issue is a duplicate of #9202 and was addressed in pull-request #9351.  In the aforementioned PR, the typedef resolution is done via `TClassEdit::ResolveTypedef()` in `GetNormalizedType()` (which is called by `RFieldBase::Create()`).

#9351 is merged; thus, closing this issue.

**Comment by mxxo** [comment]
Hey that's great! Nice one Javier :)
